bundles/framework/divmanazer/component/FilterDialog.js

```diff
@@ -627,24 +628,18 @@ Oskari.clazz.define('Oskari.userinterface.component.FilterDialog',
          * @param {Oskari.mapframework.bundle.mapwfs2.domain.WFSLayer} layer
          */
         _getLayerAttributes: function (layer) {
-            // Make copies of fields and locales
-            var fields = (layer.getFields && layer.getFields()) ? layer.getFields().slice(0) : [],
-                locales = (layer.getLocales && layer.getLocales()) ? layer.getLocales().slice(0) : [],
-                attributes = [],
-                i;
-
-            for (i = 0; i < fields.length; i += 1) {
-                // Get only the fields which originate from the service,
-                // that is, exclude those which are added by Oskari (starts with '__').
-                if (!fields[i].match(/^__/)) {
-                    attributes.push({
-                        id: fields[i],
-                        name: (locales[i] || fields[i])
+            const localizedNames = layer.getPropertyLabels();
+            if (Object.keys(localizedNames).length) {
+                // map name: locale to array containing localized names
+                return Object.entries(localizedNames)
+                    .map(([id, locale]) => {
+                        return { id, name: locale };
                     });
-                }
```

Comment:
Should we loop propertyTypes() instead and attach the matching localized name for it when found. With this approach we only show properties that have a localized name and skip any that don't.